Frameworks/Foundation/NSHTTPURLResponse.mm

```diff
@@ -14,121 +14,121 @@
 }
 
 @implementation NSHTTPURLResponse {
     NSInteger _statusCode;
-    NSMutableDictionary* _allHeaderFields;
-
-    idretain _statusLine, _statusVersion;
+    StrongId<NSMutableDictionary> _allHeaderFields;
+    StrongId<NSString> _HTTPVersion;
 }
 
 /**
  @Status Stub
 */
-+ (id)localizedStringForStatusCode:(int)code {
++ (NSString*)localizedStringForStatusCode:(int)code {
     UNIMPLEMENTED();
-    return @"HTTP error";
+    return [NSString stringWithFormat:@"Status code %d", code];
 }
 
-- (id)_initCommon:(NSURL*)url
-               statusCode:(NSInteger)statusCode
-             headerFields:(NSDictionary*)headerFields
-    expectedContentLength:(int)expectedContentLength {
-    if (self = [super init]) {
-        if (headerFields != nil) {
-            CFDictionaryKeyCallBacks caseInsensitiveKeyChecker = kCFTypeDictionaryKeyCallBacks;
+/**
+ @Status Interoperable
+*/
+- (id)initWithURL:(NSURL*)url statusCode:(NSInteger)statusCode HTTPVersion:(NSString*)HTTPVersion headerFields:(NSDictionary*)headerFields {
+    int expectedContentLength = NSURLResponseUnknownLength;
+    NSString* mimeType = nil;
+    NSString* textEncodingName = nil;
+    NSString* filename = nil;
+    NSMutableDictionary* allHeaderFields = nil;
+
+    if (headerFields != nil) {
+        CFDictionaryKeyCallBacks caseInsensitiveKeyChecker = kCFTypeDictionaryKeyCallBacks;
+
+        caseInsensitiveKeyChecker.equal = CFHTTPHeaderEqual;
+        caseInsensitiveKeyChecker.hash = CFHTTPHeaderHash;
+
+        allHeaderFields = (NSMutableDictionary*)
+            CFDictionaryCreateMutable(NULL, [headerFields count], &caseInsensitiveKeyChecker, &kCFTypeDictionaryValueCallBacks);
 
-            caseInsensitiveKeyChecker.equal = CFHTTPHeaderEqual;
-            caseInsensitiveKeyChecker.hash = CFHTTPHeaderHash;
+        //  Case insensitive dictionary
+        for (id key in [headerFields allKeys]) {
+            id val = [headerFields objectForKey:key];
+            [allHeaderFields setObject:val forKey:key];
+        }
+
+        NSString* expectedLength = [allHeaderFields objectForKey:@"Content-Length"];
+        if (nil != expectedLength) {
+            expectedContentLength = [expectedLength intValue];
+        }
 
-            _allHeaderFields = (NSMutableDictionary*)
-                CFDictionaryCreateMutable(NULL, [headerFields count], &caseInsensitiveKeyChecker, &kCFTypeDictionaryValueCallBacks);
+		// Parse the filename from the Content-Disposition header field.
+        NSString* contentDisposition = [allHeaderFields objectForKey:@"Content-Disposition"];
+        NSRange filenameTagPosition = [contentDisposition rangeOfString:@"filename=" options:NSBackwardsSearch];
+        NSCharacterSet* filenameEndDelimiter;
+
+        if (filenameTagPosition.location != NSNotFound) {
+            NSUInteger startPos = NSMaxRange(filenameTagPosition); // denotes the start position of the file name
+
+			// if a quote follows 'filename=', then use the entire quoted string
```

Comment:
nooooooooooooooooooooooooooo

---
In reply to: [69209618](https://github.com/Microsoft/WinObjC/pull/595#discussion_r69209618) [](ancestors = 69209618)